unc_classification_tab_data/utils/visualizing_utils.py

```diff
@@ -0,0 +1,211 @@
+            nth list of labels in y.
+        unc_dict: Dict[List[np.ndarray]]
+            The same as y_pred_dict, except that it contains the predicted uncertainties instead
+            of the predicted probabilities.
+        metrics: List[function]
+            A list of metrics that we want to calculate. All metrics should take labels and
+            probabilities in this order: f(y, y_pred).
+        min_size: int
+            The minimum size of the dataset for which we want to evaluate the metrics.
+        step_size: int
+            The number of items that we want to add at each increment.
+        """
+        self.y = y
+        self.y_pred = y_pred_dict
+        self.uncertainty_dict = unc_dict
+        self.loss_dict: defaultdict = defaultdict(list)
+        self.loss_std_dict: defaultdict = defaultdict(list)
+        self.xs = dict()
+        self.metrics = metrics
+        self.min_size = min_size
+        self.step_size = step_size
+        self._calculate_incremental_metrics(metrics)
+
+    def plot_incremental_metric(self, metric, title=None, methods=None, alpha=0.1,
+                                key_mapping=None, legend=True):
+        """Plot how a metric changes when adding more uncertain points. Do this for multiple
+        methods, such as Bayesian NN and KNN.
+
+        Parameters
+        ----------
+        metric: str
+            The name of the metric (metric.__name__).
+        title: str (optional)
+            The title of the plot.
+        methods: List[str] (optional)
+            A list with names of methods for which we want to see the metric (e.g. "KNN").
+        alpha: float
+            The shade of the standard deviation in the plot.
+        key_mapping: dict (optional)
+            A dictionary that maps method names to "prettier" names for display in the plot.
+        """
+
+        sns.set_style('whitegrid')
+        if not methods:  # if no methods are specified, plot all.
+            methods = self.uncertainty_dict.keys()
+        sns.set_palette("Set1", len(methods))
+        marker = itertools.cycle(('X', 'o', '^', 'v', 'd', '*', '.'))
+        for m in methods:
+            x_values = np.array(self.xs[m][metric]) / len(self.y[0])
+            met = np.array(self.loss_dict[m][metric])
+            std = np.array(self.loss_std_dict[m][metric])
+            if key_mapping is not None:
+                label = key_mapping[m]
+            else:
+                label = m
+            plt.plot(x_values, met, label=label, marker=next(marker), markersize=8)
+            plt.fill_between(x_values,
+                             met - std,
+                             met + std,
+                             alpha=alpha)
+        plt.xlabel("Fraction of included data points \n(most certain points are included first)")
+        if legend:
+            plt.legend()
+        if title:
+            plt.title(title)
+        else:
+            plt.title(metric)
+        plt.tight_layout()
```

Comment:
maybe return the figure if you want to save it?

Reply:
I didn't do it for now, I was not sure why it is useful in this case?

Reply:
because someone might want to save it in the future :)